x-pack/auditbeat/processors/sessionmd/processdb/db.go

```diff
@@ -712,29 +709,6 @@ func getTTYType(major uint16, minor uint16) TTYType {
 	return TTYUnknown
 }
 
-func (db *DB) scrapeAncestors(proc Process) {
```

Comment:
we don't need `scrapeAncestors` any more?

Reply:
no, it's not needed. I removed this for two reasons:

I thought it would help when I added it, but now I've never seen any evidence that it actually does. The changes I made here fixed all the problems I had with ebpf, and I haven't seen it used at all. When processes are missed with procfs backend the process will have ended before it's scraped, so it never helped.

I think it's better that the enrich function be read-only. It's a bit strange that the enrich function was actually writing to the DB sometimes (and it tripped me up when debugging)

There is still procfs scraping with the ebpf backend, when starting up to fill in processes that started before the processor, but that separate

Reply:
ok now I see thanks for the explanation @mjwolf